typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -261,87 +283,193 @@ export class EvmERC20WarpRouteReader extends HyperlaneReader {
     type: TokenType,
     warpRouteAddress: Address,
   ): Promise<HypTokenConfig> {
-    if (
-      type === TokenType.collateral ||
-      type === TokenType.collateralVault ||
-      type === TokenType.collateralVaultRebase ||
-      type === TokenType.XERC20 ||
-      type === TokenType.XERC20Lockbox
-    ) {
-      let xerc20Token: Address | undefined;
-      let lockbox: Address | undefined;
-      let token: Address;
-      let xERC20Metadata: XERC20TokenMetadata | {} = {};
-
-      if (type === TokenType.XERC20Lockbox) {
-        // XERC20Lockbox is a special case of collateral, we will fetch it from the xerc20 contract
-        const hypXERC20Lockbox = HypXERC20Lockbox__factory.connect(
-          warpRouteAddress,
-          this.provider,
-        );
-        xerc20Token = await hypXERC20Lockbox.xERC20();
-        token = xerc20Token;
-        lockbox = await hypXERC20Lockbox.lockbox();
-      } else {
-        const erc20 = HypERC20Collateral__factory.connect(
-          warpRouteAddress,
-          this.provider,
-        );
-        token = await erc20.wrappedToken();
-      }
+    // Using null instead of undefined to force
+    // a compile error when adding a new token type
+    const mapConfig: Record<
```

Comment:
This map is the same every time. Why not make it an instance variable, and initialize in the constructor?